packages/iov-core/src/signingservercore.spec.ts

```diff
@@ -228,4 +229,41 @@ describe("SigningServerCore", () => {
 
     core.shutdown();
   });
+
+  it("can handle rejected sign and post request", async () => {
+    pendingWithoutBnsd();
+
+    const profile = new UserProfile();
+    const signer = new MultiChainSigner(profile);
+    const { connection } = await signer.addChain(bnsConnector(bnsdUrl));
+    const bnsChain = connection.chainId();
+
+    {
+      const wallet = profile.addWallet(
+        Ed25519HdWallet.fromMnemonic(
+          "option diagram plastic million educate they arrow fat comic excite abandon green",
+        ),
+      );
+      await profile.createIdentity(wallet.id, bnsChain, HdPaths.simpleAddress(1));
+    }
+
+    async function rejectAllTransactions(_1: string, _2: UnsignedTransaction): Promise<boolean> {
+      return false;
+    }
+
+    const core = new SigningServerCore(profile, signer, defaultGetIdentitiesCallback, rejectAllTransactions);
+
+    const identities = await core.getIdentities("Please select signer", [bnsChain]);
+    const signingIdentity = identities[0];
+    const send: SendTransaction = {
+      kind: "bcp/send",
+      creator: signingIdentity,
+      amount: defaultAmount,
+      recipient: await randomBnsAddress(),
+    };
+    const transactionId = await core.signAndPost("Please sign now", send);
+    expect(transactionId).toBeUndefined();
```

Comment:
Okay, so we check for undefined in the client as rejection. An error would be like a failed network connection, right?

Reply:
Right